### PR TITLE
[ENG-5337] Dep-group: external services (dropbox / box / etc.)

### PR DESCRIPTION
**Pull request by opaduchak**
Bumped external services dependencies
## Purpose

Python and libs upgrade

## Changes

Bumped libraries, adapted existing code to newer version of libraries

## Ticket

[ENG-5337](https://openscience.atlassian.net/browse/ENG-5337)

[ENG-5337]: https://openscience.atlassian.net/browse/ENG-5337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by opaduchak** [comment]
With update to arrow>=0.15.0 mendeley may be broken (arrow.get method changed spec in some cases)

File: addons/gitlab/api.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 from addons.gitlab.settings import DEFAULT_HOSTS
 
 # Initialize caches
-https_cache = cachecontrol.CacheControlAdapter()
-default_adapter = HTTPAdapter()
+https_cache = cachecontrol.CacheControlAdapter()  # isn't used anywhere, maybe delete?
+default_adapter = HTTPAdapter()  # isn't being used anywhere, maybe delete?
```
Comment (felliott) [inline addons/gitlab/api.py:15]: Well spotted!  A lot of our addons began life as a cloned version of another addon.  This one was cloned from the github addon, which depends on (and uses) `cachecontrol` and `python-magic`.  But the gitlab addon never used those modules.  I will delete this code and the deps from gitlab's requirements.txt

File: addons/gitlab/views.py [hunk 2]
```diff
@@ -255,6 +255,7 @@ def gitlab_root_folder(*args, **kwargs):
 # Repos #
 #########
 
+# gitlab parameter shadows import statement
```
Comment (felliott) [inline addons/gitlab/views.py:258]: Good eyes! The `gitlab` param would be better named as `node_settings`. I will update this.

**Comment by felliott** [comment]
Added some minor updates, but otherwise looks good! Merged.

**Pull request merged**